### DISCUSSION TITLE
Different controller behaviour in v7.41.0?

### DISCUSSION THREAD

**Discussion by manuelpoelzl** [react-hook-form/react-hook-form#9613] [Q&A]
Hi guys,

i just stumbled across some weird behaviour with the new version v7.41.0.
It seems like the defaultValue of the controller can no longer be accessed with the field.value property.

in my project, i have a function, which renders my form fields depending on the settings i get from the database:
```jsx
const ReturnAndRenderSettings = (): ReactNode[] => {
        return availableSettings.map((s) => {
            return (
                <div className={"field"}>
                    <Controller name={s.Name} control={settingsControl}
                                defaultValue={userSettings.find(
                                    us => us.SettingId == s.ID)?.IsEnabled ?? false}
                                render={({field, fieldState}) => (

                                    <Setting name={s.Name}
                                             disabled={inSubmit}
                                             defaultValue={field.value}
                                             onChange={setSettingsValue}/>
                                )}/>
                </div>
            );
        })
    };
```

As you can see here, i´m searching for the matching setting which i get from the server, and then i get the actual value from the ```IsEnabled``` property. As fallback i´m using false.

This code worked fine till v7.41.0 and the result looked like this:
![Screenshot_1](https://user-images.githubusercontent.com/46539758/208323938-08ec0c5e-7457-447e-838e-87afc5f1d3f1.png)

However with the newst version ```field.value``` is always ```undefined```.
The result, as expected, looks like this:
![Screenshot_2](https://user-images.githubusercontent.com/46539758/208324053-df5ef5dc-7225-47eb-8d50-43c354c71cef.png)

It seems like in version <7.41.0 field.value was identical with the controller default value, and since 7.41.0 this is not the case anymore.

I´ve read about the new values and defaultValues things in the new versions but found no connection to the controller, so this makes me wonder if i´m missing something here🤔

**Comment by bluebill1049** [comment]
any chance you could try to reproduce this in an isolated codesandbox? we can try to find out what's causing the issue and get it fixed.

**Reply by bluebill1049** [reply]
here is the release note: https://github.com/react-hook-form/react-hook-form/releases/tag/v7.41.0 i didn't spot anything closely related to useController/controller

**Reply by manuelpoelzl** [reply]
Thanks for the quick response!
I was able to reproduce the issue within a code sandbox sample ([CodeSandbox](https://codesandbox.io/s/shy-wind-pvwc0d?file=/src/App.tsx)).

The sample is currently targeting react-hook-form v7.39.4 .
If you open it, you´ll see, that the first switch is active.
![image](https://user-images.githubusercontent.com/46539758/208417424-de25e282-6483-405b-831a-7f6d47772846.png)

But after you switch to the latest version of react-hook-form (7.41.0) you´ll see that the switch is not active anymore
![image](https://user-images.githubusercontent.com/46539758/208417581-ff999505-7136-4a66-8163-ddb6ad9eed62.png)

**Reply by bluebill1049** [reply]
thanks logged an issue: https://github.com/react-hook-form/react-hook-form/issues/9621

**Reply by manuelpoelzl** [reply]
Thank you for looking into it!
